qi-test/tests/private/util.rkt

```diff
@@ -54,3 +55,33 @@
 (define-qi-foreign-syntaxes double-me)
 
 (define-qi-foreign-syntaxes add-two)
+
+(define (syntax-list? v)
+  (and (syntax? v) (syntax->list v)))
+
+(define (tree-map f tree)
+  (cond [(list? tree) (map (curry tree-map f)
+                           tree)]
+        [(syntax-list? tree) (f (datum->syntax tree
+                                  (tree-map f (syntax->list tree))))]
```

Comment:
I suspect we should be passing the other two optional arguments to `datum->syntax` here, but since it's only for tests it _might_ not be relevant?

Reply:
Interesting, I didn't realize that properties could be automatically propagated by passing the input syntax for the `prop` argument. That is very useful to know. I'll make this change in a follow-up PR, thank you.